src/common/tests/policy_tests.c

```diff
@@ -0,0 +1,224 @@
+	assert_true(are_policy_descs_equal(&out, &exp_out));
+
+}
+
+static void
+test_policy_no_type(void **state)
+{
+	const char *str = "th1=6";
+	struct policy_desc_t out, exp_out;
+	bool result = false;
+
+	reset_policy_desc(&out);
+	reset_policy_desc(&exp_out);
+	result = daos_policy_try_parse(str, &out);
+
+	exp_out.policy = DAOS_MEDIA_POLICY_WRITE_INTENSIVITY;
+
+	assert_false(result);
+}
+
+static void
+test_policy_blank(void **state)
+{
+	const char *str = "";
+	struct policy_desc_t out, exp_out;
+	bool result = false;
+
+	reset_policy_desc(&out);
+	reset_policy_desc(&exp_out);
+	result = daos_policy_try_parse(str, &out);
+
+	exp_out.policy = DAOS_MEDIA_POLICY_WRITE_INTENSIVITY;
+
+	assert_false(result);
+
+}
+
+static void
+test_policy_junk(void **state)
+{
+	const char *str = "dfgj=jaosdfhg/asdg=2346/wgdsh=25";
+	struct policy_desc_t out, exp_out;
+	bool result = false;
+
+	reset_policy_desc(&out);
+	reset_policy_desc(&exp_out);
+	result = daos_policy_try_parse(str, &out);
+
+	exp_out.policy = DAOS_MEDIA_POLICY_WRITE_INTENSIVITY;
+
+	assert_false(result);
+
+}
+
+
+static void
+test_policy_too_many_params(void **state)
+{
+	const char *str = "type=io_size/th1=512/th2=4096/th3=666/th4=42/th5=6";
+	struct policy_desc_t out, exp_out;
+	bool result = false;
+
+	reset_policy_desc(&out);
+	reset_policy_desc(&exp_out);
+	result = daos_policy_try_parse(str, &out);
+
+	assert_false(result);
+
+}
+
+static void
+test_policy_bad_param(void **state)
+{
+	const char *str = "type=io_size/th1=asdf/th2=4096";
+	struct policy_desc_t out, exp_out;
+	bool result = false;
+
+	reset_policy_desc(&out);
+	reset_policy_desc(&exp_out);
+	result = daos_policy_try_parse(str, &out);
+
+	assert_false(result);
+}
+
+static void
+test_policy_bad_chars(void **state)
+{
+	const char *str = "Q$=%,*%#^*($^&RGFH";
+	struct policy_desc_t out, exp_out;
+	bool result = false;
+
+	reset_policy_desc(&out);
+	reset_policy_desc(&exp_out);
+	result = daos_policy_try_parse(str, &out);
+
+	assert_false(result);
+}
+
+static void
+test_policy_type_not_first(void **state)
+{
+	const char *str = "th1=512/th2=4096/type=io_size";
+	struct policy_desc_t out, exp_out;
+	bool result = false;
+
+	reset_policy_desc(&out);
+	reset_policy_desc(&exp_out);
+	result = daos_policy_try_parse(str, &out);
+
+	exp_out.policy = DAOS_MEDIA_POLICY_IO_SIZE;
+	exp_out.params[0] = 512;
+	exp_out.params[1] = 4096;
+
+	assert_false(result);
+}
+
+int
+main(void)
+{
+	const struct CMUnitTest tests[] = {
+		cmocka_unit_test(test_policy_positive),
+		cmocka_unit_test(test_policy_negative),
+		cmocka_unit_test(test_policy_type_only),
+		cmocka_unit_test(test_policy_no_type),
+		cmocka_unit_test(test_policy_blank),
+		cmocka_unit_test(test_policy_junk),
+		cmocka_unit_test(test_policy_too_many_params),
+		cmocka_unit_test(test_policy_bad_param),
+		cmocka_unit_test(test_policy_bad_chars),
+		cmocka_unit_test(test_policy_type_not_first),
+	};
+
+	return cmocka_run_group_tests_name("policy_tests", tests, NULL, NULL);
+}
```

Comment:
Are those tests passing? Shall we integrate them in the unit test?

Reply:
I run test manually, it passed test.
[==========] Running 10 test(s).
[ RUN      ] test_policy_positive
[       OK ] test_policy_positive
[ RUN      ] test_policy_negative
[       OK ] test_policy_negative
[ RUN      ] test_policy_type_only
[       OK ] test_policy_type_only
[ RUN      ] test_policy_no_type
[       OK ] test_policy_no_type
[ RUN      ] test_policy_blank
[       OK ] test_policy_blank
[ RUN      ] test_policy_junk
[       OK ] test_policy_junk
[ RUN      ] test_policy_too_many_params
[       OK ] test_policy_too_many_params
[ RUN      ] test_policy_bad_param
[       OK ] test_policy_bad_param
[ RUN      ] test_policy_bad_chars
[       OK ] test_policy_bad_chars
[ RUN      ] test_policy_type_not_first
[       OK ] test_policy_type_not_first
[==========] 10 test(s) run.
[  PASSED  ] 10 test(s).

Reply:
There are kind of tests like this, but they are not integrated into tests automatically, i supposed these are used by developers? maybe one test missed here could be function test, eg set different throttle size other than default, make sure
IO write to PMEM or NVME properly which could be pushed as separate PR if you are ok.